sound/soc/intel/boards/sof_sdw.c

```diff
@@ -599,6 +599,14 @@ static const struct dmi_system_id sof_sdw_quirk_table[] = {
 		},
 		.driver_data = (void *)(SOC_SDW_CODEC_SPKR),
 	},
+	{
+		.callback = sof_sdw_quirk_cb,
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "LENOVO"),
+			DMI_EXACT_MATCH(DMI_PRODUCT_SKU, "233C")
+		},
+		.driver_data = (void *)(SOC_SDW_CODEC_MIC),
```

Comment:
nitpick: I would recommend adding a comment to say the codec MIC is excluded. Because the quirk usually means included.